File: foodtruck-api/foodtruck-api/src/main/java/com/foodtruck/foodtruckapi/service/UserService.java
package com.foodtruck.foodtruckapi.service;

import com.foodtruck.foodtruckapi.dto.request.CreateUserRequest;
import com.foodtruck.foodtruckapi.dto.request.UpdateUserRequest;
import com.foodtruck.foodtruckapi.dto.response.UserResponse;

import java.util.List;

public interface UserService {
    UserResponse createUser(CreateUserRequest userRequest);
    UserResponse updateUser(Integer id, UpdateUserRequest userRequest);
    List<UserResponse> getAllUsers();
    UserResponse getUserById(Integer id);
    UserResponse deactivateUser(Integer id);
}

File: frontend/src/services/userService.ts
import api from './api';
import type {User, UserResponse, CreateUserRequest, UpdateUserRequest} from '../types/user.types'

export const getAllUsers = async () => {
    const response = await api.get<UserResponse[]>('/users');
    return response.data;
}

export const getUserById = async (id: number) => {
    const response = await api.get<UserResponse>(`/users/${id}`);
    return response.data;
}

export const createUser = async (data: CreateUserRequest) :Promise<User> => {
    const response = await api.post<User>('/users', data);
    return response.data;
}

export const updateUser = async (id:number, data: UpdateUserRequest): Promise<User> =>{
    const response = await api.put<User>(`/users/${id}`, data);
    return response.data;
}

export const deactivateUser = async (id:number) => {
    const response = await api.put<User>(`/users/${id}/deactivate`);
    return response.data;
}